site/documentation/content/api/credentials/index.md

```diff
@@ -302,6 +302,39 @@ Example:
+The example above does not contain any of the `not-before`, `not-after` and `enabled` properties. The `not-before`
+and `not-after` properties should be omitted if the validity period is the same as the period indicated by the client
+certificate's corresponding properties. It is still necessary to provide a (empty) JSON object in the *secrets* array,
+though.
 {{% /notice %}}
 
+### Raw Public Key
+
+A credentials type for storing a public key as used in the [JSON Web Token based Authentication]({{< relref "concepts/device-identity.md#json-web-token-based-authentication" >}}) to authenticate a device.
+
+Example:
+
+~~~json
+{
+  "device-id": "4711",
+  "type": "rpk",
+  "auth-id": "sensor1",
+  "secrets": [
+    {
+      "algorithm": "RSA",
+      "key": "-----BEGIN PUBLIC KEY-----MIIBIj...IDAQAB-----END PUBLIC KEY-----"
+    }
+  ]
+}
+~~~
+
+| Name        | Mandatory | JSON Type | Description                                                                                                                                                                                                                                                                                                |
+|:------------|:---------:|:----------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| *type*      |   *yes*   | *string*  | The credential type name, always `rpk`.                                                                                                                                                                                                                                                                    |
+| *auth-id*   |   *yes*   | *string*  | The identity that the device should be authenticated as.                                                                                                                                                                                                                                                   |
+| *algorithm* |   *yes*   | *string*  | The signing algorithm of the provided key (supported algorithms are `RSA` and `EC`).                                                                                                                                                                                                                       |
+| *key*       |   *yes*   | *string*  | The Base64 encoded bytes representing the public key / certificate wrapped in a header and footer. Header and footer (either `-----BEGIN PUBLIC KEY-----` and `-----END PUBLIC KEY-----` for public keys or `-----BEGIN CERTIFICATE-----` and `-----END CERTIFICATE-----` for certificates) are mandatory. |
```

Comment:
This property does only contain the Base64 encoding of the DER encoded public key. There are no -----BEGIN PUBLIC KEY----- headers/footers ...

Reply:
updated, but this can actually also be an encoded certificate